src/minio/core/minio_client.py

```diff
@@ -81,3 +86,213 @@ async def test_connection(self) -> bool:
+
+        Returns:
+            True if the bucket and its contents were successfully deleted.
+
+        Raises:
+            BucketOperationError: If the bucket deletion fails.
+        """
+        try:
+            async with self._get_client() as client:
+                # Always delete all objects first
+                paginator = client.get_paginator("list_objects_v2")
+                async for page in paginator.paginate(Bucket=bucket_name):
+                    if "Contents" in page:
+                        objects = [{"Key": obj["Key"]} for obj in page["Contents"]]
+                        await client.delete_objects(
+                            Bucket=bucket_name, Delete={"Objects": objects}
+                        )
+
+                await client.delete_bucket(Bucket=bucket_name)
+                logger.info(f"Deleted bucket: {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to delete bucket {bucket_name}: {e}")
+            raise BucketOperationError(f"Bucket deletion failed: {e}") from e
+
+    async def put_object(self, bucket_name: str, key: str, body: bytes) -> bool:
+        """
+        Uploads an object to a specified bucket.
+
+        Args:
+            bucket_name: The name of the target bucket.
+            key: The object key (i.e., its name/path within the bucket).
+            body: The content of the object as bytes.
+
+        Returns:
+            True if the object was uploaded successfully.
+
+        Raises:
+            MinIOManagerError: If the upload operation fails.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.put_object(Bucket=bucket_name, Key=key, Body=body)
+                logger.info(f"Put object {key} in bucket {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to put object {key} in bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object put failed: {e}") from e
+
+    async def get_object(self, bucket_name: str, key: str) -> bytes:
+        """
+        Retrieves an object from a bucket.
+
+        Args:
+            bucket_name: The name of the bucket containing the object.
+            key: The key of the object to retrieve.
+
+        Returns:
+            The content of the object as bytes.
+
+        Raises:
+            MinIOManagerError: If the object retrieval fails.
+        """
+        try:
+            async with self._get_client() as client:
+                response = await client.get_object(Bucket=bucket_name, Key=key)
+                body = await response["Body"].read()
+                logger.info(f"Got object {key} from bucket {bucket_name}")
+                return body
+        except Exception as e:
+            logger.error(f"Failed to get object {key} from bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object get failed: {e}") from e
```

Comment:
Same questions, more or less, as the above

Reply:
It's also for the service to use as a util function - I don't have a plan to build an endpoint associated with this.

Reply:
I'd just add docs saying that getting large objects via this method isn't a good idea

Reply:
👍

Reply:
:+1: